packages/client/src/components/pages/muster-page/table-unit-muster-compliance.tsx

```diff
@@ -61,13 +62,11 @@ export const UnitMusterComplianceTable = (props: any) => {
   const [rosterFromDateIso, setRosterFromDateIso] = useState(props.rosterFromDateIso);
   const [rosterToDateIso, setRosterToDateIso] = useState(props.rosterToDateIso);
   const [rosterUnitId, setRosterUnitId] = useState(props.rosterUnitId);
-  const [unitComplianceData, setUnitComplianceData] = useState<Map<string, ComplianceData>>(new Map());
+  const [unitComplianceStats, setUnitComplianceStats] = useState<Map<string, ComplianceStats>>(new Map());
```

Comment:
Is there any benefit to using Map here instead of a plain object since the keys are just strings? I guess seeing Map implies to me that something unusual is going on with non-string keys, but this seems to be doing the same thing as a regular object.

Reply:
We get the typing on the values also. Otherwise, how do I define an object where every value of a property/"key" must conform to a certain type?

Reply:
You should be able to do this to get the same level of typing with plain objects.

```typescript
const [unitComplianceStats, setUnitComplianceStats] = useState<{ [key: string]: ComplianceStats }>({});
```

Reply:
Or, alternatively, like this (though I generally prefer the first method since you can add context with a key name).

```typescript
const [unitComplianceStats, setUnitComplianceStats] = useState<Record<string, ComplianceStats>>({});
```

Reply:
Interesting I have never seen that notation for objects before. I gave it a try just for my own learning and the weird thing is that the value type is enforced but `[key: string]` seems to behave oddly. Like I can totally do the following (perhaps toString is called internally on number? idk seems sketchy though):
```
     const test: { [key: string]: string } = {};
      test[1] = 'whatev';
      test[2] = 'whatev';
 ```

Reply:
Yeah JS automatically converts numbers to strings in object keys. It would be nice if TS threw an error for that. Maybe there's some way to enforce that in the tsconfig.

It'll at least throw errors for union types, if you do something like this...

```typescript
type Things = 'a' | 'b';

const test: { [K in Things]?: number } = {};
test['a'] = 1; // ok
test['c'] = 2; // error
test[1] = 3; // error
```